build.rs

```diff
@@ -116,9 +183,15 @@ fn find_package(name: &str) -> Vec<PathBuf> {
 fn find_package(name: &str) -> Vec<PathBuf> {
     if let Ok(vcpkg_root) = std::env::var("VCPKG_ROOT") {
         vec![link_vcpkg(vcpkg_root.into(), name)]
-    } else {
-        // Try using homebrew
+    } else if is_macports_installed() {
+        // Try using MacPorts
+        vec![link_package_m1(name, PathBuf::from("/opt/local/include"), PathBuf::from("/opt/local/lib"))]
+    } else if is_homebrew_installed() {
+        // Try using Homebrew
         vec![link_homebrew_m1(name)]
+    } else {
+        // Local installation
+        vec![link_package_m1(name, PathBuf::from("/usr/local/include"), PathBuf::from("/usr/local/lib"))]
```

Comment:
SInce is_macports_installed use the local installation path, why borther adding is_macports_installed?

Reply:
> SInce is_macports_installed use the local installation path, why borther adding is_macports_installed?

The user could install macports then uninstall, the files remain, but the "port" executable is not installed/removed, so using macports is not relevant on the target system. Similarly with Homebrew.

Reply:
> You make macports having higher prioirity than brew, it is not fair. Brew is more widely used.

Well, I write code not from the principles of fairness, but from the principle of convenience. I use MacPorts and it's more native to Unix/Linux systems than Homebrew.

Reply:
> You make macports having higher prioirity than brew, it is not fair. Brew is more widely used.

And if it was only about Homebrew, then by actually adding support for Homebrew, you do not take into account the option that the user will not be able to use tools such as Homebrew or MacPorts at all, and the latter will be completely compatible with their own directory structure.

Reply:
Then why borther check if macport installed instead of check if something like `/usr/local/include/opus.h` exists which it is more native than your way.

Reply:
> but `/opt/local/include` and `/usr/local/include` are hard coded. What if the guy installed macports, but not install opus via macports, but install under /user/local by himself

Yeah, hardcoded such as this: `/opt/homebrew/Cellar`

Reply:
I am not perfectionist. But I know if the logic is perfect, we should give homebrew higher priority, we should respect existing code.